src/rust/ide/src/view/node_editor.rs

```diff
@@ -699,26 +718,46 @@ impl GraphEditorIntegratedWithControllerModel {
         Ok(())
     }
 
+    fn expression_entered_in_ui
+    (&self, local_call:&Option<LocalCall>) -> FallibleResult<()> {
+        if let Some(local_call) = local_call {
+            let method_pointer = local_call.definition.clone();
+            let expression_id  = local_call.call;
+            let controller     = self.controller.clone_ref();
+            let logger         = self.logger.clone_ref();
+            let enter_action   = async move {
+                info!(logger,"Entering node.");
+                let result = controller.enter_method_pointer(expression_id,&method_pointer).await;
+                if let Err(e) = result {
```

Comment:
Here and before I'd prefer either to keep original log — that was occuring regardless whether entering succeeded or not. That way (by comparing it against the previous one) we could say that entering handling code started here and ended there.

If you want prettier messages, I'm fine with pattern matching result and pretty-printing more specific logs in both cases (using error level in case of failure). I just don't want to end up with less information in logs here.